### PR TITLE
Fix upstream to include `bool` type for `Index`

### LINKED ISSUE #8722: [upstream/build] Groupby mean result index doesn't preserve bool dtype
**What happened**:

@ncclementi and I were tracking down this error on her fork with upstream dependencies:
https://github.com/ncclementi/dask/runs/5148303425?check_suite_focus=true#step:5:11490

Following a recent pandas bug enhancement (https://github.com/pandas-dev/pandas/pull/45061), groupby results with bool-typed indexes are better respected. In dask, the index type generally remains `bool`, but `groupby.mean()` returns a dataframe with an `object`-dtyped index.

If you look at the example below, most operations return a bool index. Mean returns an object index. Interestingly, doing mathematical operations with two dataframes with bool-typed indexes returns a dataframe with an object-typed index

cc: @jrbourbeau 
We were not able to track down where the fix for this would occur and need some guidance.

**What you expected to happen**:

`Groupby.mean` should preserve the index dtypes, which in turn will allow the tests to pass 

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
import dask.dataframe as dd

pdf = pd.DataFrame(
            {"a": [1, 2, 6, 4, 4, 6, 4, 3, 7], "b": [4, 2, 7, 3, 3, 1, 1, 1, 2]},
            index=[0, 1, 3, 5, 6, 8, 9, 9, 9],
        )
ddf = dd.from_pandas(pdf, npartitions=3)

gb = ddf.groupby(ddf.b > 4).a

assert gb.count().index.dtype == bool  # good
assert gb.sum().index.dtype == bool  # good
assert gb.mean().index.dtype == object  # bad (implemented as sum() / count(), see link below)
## and also:
assert (gb.sum() / gb.count()).index.dtype == object  # bad

```

https://github.com/dask/dask/blob/d670db6f58e47e8b0a5019f75330121dda11af84/dask/dataframe/groupby.py#L1424-L1429

**Anything else we need to know?**:

**Environment**:

- Dask version: main (d670db6f58e47e8b0a5019f75330121dda11af84)
- Pandas: upstream (1.5.0.dev0+350.g0631637205)
- Python version: 3.9.10
- Operating System: Mac OS 12.2.1 (21D62)
- Install method (conda, pip, source): source

<details>
<summary>Cluster Dump State:</summary>

</details>

### PR DESCRIPTION
**Pull request by ncclementi**
- [x] Closes #8722
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

There was pandas bug enhancement (https://github.com/pandas-dev/pandas/pull/45061), that includes the possibility of a boolean Index.

In dask operations that involve `/` or `truediv` (see comment https://github.com/dask/dask/issues/8722#issuecomment-1040734933) returns a dataframe with an object-dtyped index instead. 

This PR includes includes the `idx.dtype` when registering the meta for the case of `pd.Index`

### CODE REVIEW

**Comment by ncclementi** [comment]
It looks like the fix affected another set of tests. `test_dataframe_aggregations_multilevel` the cases that involve `mean` which makes sense as mean involves a `/` operation. See https://github.com/dask/dask/runs/5233914770?check_suite_focus=true#step:5:10818

**Comment by jsignell** [comment]
I took a look at the git blame and that "boolean != empty" goes back to https://github.com/dask/dask/pull/1692 I would not be surprised if that needed to change. I can't quite tell what it was trying to get at, but fundamentally the dtype of the index should not depend on the length of the index.

**Comment by ncclementi** [comment]
This looks green! The only tests failing on the upstream build are being taken care of in https://github.com/dask/dask/pull/8558

@jsignell for the "boolean !=empty" message we ended telling it to NOT `XFAIL` when having `pandas >=1.5`. Unless you think there is something else that should be taken care of regarding that choice, I think this one is ready for a final review.

File: dask/dataframe/backends.py [hunk 1]
```diff
@@ -194,7 +194,13 @@ def _nonempty_index(idx):
     elif idx.is_numeric():
         return typ([1, 2], name=idx.name)
     elif typ is pd.Index:
-        return pd.Index(["a", "b"], name=idx.name)
+        if idx.dtype == bool:
+            # pd 1.5 introduce bool dtypes and respect non-uniqueness
+            return pd.Index([True, False], name=idx.name)
+        else:
+            # for pd 1.5 in the case of bool index this would be cast as [True, True]
+            # braking uniqueness
```
Comment (jsignell) [inline dask/dataframe/backends.py:202]: nit: typo
```suggestion
            # breaking uniqueness
```

**Comment by jsignell** [comment]
Thanks @ncclementi! This is in.

**Pull request merged**